hazelcast/src/main/java/com/hazelcast/map/EntryProcessor.java

```diff
@@ -85,22 +95,35 @@
      * </pre>
      * otherwise the {@code EntryProcessor} does not guarantee to modify the entry.
      * <p>
-     * The entry's value will be {@code null}, if the entry for the key doesn't exist. You
-     * can create new mapping by setting a non-null value or remove existing mapping
-     * entry by setting the value to null.
+     * The entry's value will be {@code null}, if the entry for the key doesn't
+     * exist. You can create new mapping by setting a non-null value or remove
+     * existing mapping entry by setting the value to {@code null}.
      *
      * @param entry entry to be processed
-     * @return a result that will be returned from the method taking the {@link EntryProcessor}, such as
+     * @return a result that will be returned from the method taking the
+     * {@link EntryProcessor}, such as
      * {@link com.hazelcast.core.IMap#executeOnKey(Object, EntryProcessor) IMap.executeOnKey()}
      */
-    Object process(Map.Entry<K, V> entry);
+    R process(Entry<K, V> entry);
 
     /**
      * Get the entry processor to be applied to backup entries.
+     * <p>
+     * In case of a readonly execution, {@code null} can be returned to indicate
+     * that no backups should be made.
+     * <p>
+     * Note that there is a possibility which an {@code EntryProcessor} can see
+     * that a key exists but its backup processor may not find it at run time due
+     * to an unsent backup of a previous operation (e.g. a previous put). In
+     * those situations, Hazelcast internally/eventually will sync those owner
+     * and backup partitions so you will not lose any data.
+     * When coding an backup entry processor, you should take that case into
+     * account, otherwise {@link java.lang.NullPointerException}s can be seen
+     * since {@link java.util.Map.Entry#getValue()} may return null.
```

Comment:
Maybe add a line to say return value of backup entry processor is ignored. People may get confused as they do not know what should be returned from a backup entry processor.